Common/include/geometry/CPhysicalGeometry.hpp

```diff
@@ -38,16 +38,17 @@
  */
 class CPhysicalGeometry final : public CGeometry {
 
-  map<unsigned long, unsigned long> Global_to_Local_Point;  /*!< \brief Global-local indexation for the points. */
-  long *Local_to_Global_Point;                              /*!< \brief Local-global indexation for the points. */
-  unsigned short *Local_to_Global_Marker;                   /*!< \brief Local to Global marker. */
-  unsigned short *Global_to_Local_Marker;                   /*!< \brief Global to Local marker. */
-  unsigned long *adj_counter;                               /*!< \brief Adjacency counter. */
-  unsigned long **adjacent_elem;                            /*!< \brief Adjacency element list. */
-  su2activematrix Sensitivity;                              /*!< \brief Matrix holding the sensitivities at each point. */
+  unordered_map<unsigned long, unsigned long>
+  Global_to_Local_Point;                    /*!< \brief Global-local indexation for the points. */
+  long *Local_to_Global_Point;              /*!< \brief Local-global indexation for the points. */
+  unsigned short *Local_to_Global_Marker;   /*!< \brief Local to Global marker. */
+  unsigned short *Global_to_Local_Marker;   /*!< \brief Global to Local marker. */
+  unsigned long *adj_counter;               /*!< \brief Adjacency counter. */
+  unsigned long **adjacent_elem;            /*!< \brief Adjacency element list. */
+  su2activematrix Sensitivity;              /*!< \brief Matrix holding the sensitivities at each point. */
 
   vector<vector<unsigned long> > Neighbors;
-  map<unsigned long, unsigned long> Color_List;
+  unordered_map<unsigned long, unsigned long> Color_List;
```

Comment:
Moving down the list of geometry preprocessing bottlenecks, accessing std::map items is O(log(N)), std::unordered_map is O(1).
Why do I care about preprocessing you ask? Because it is not (as) easy to make hybrid parallel, and so it better be quick.

Reply:
Agreed - it is generally better performance with unordered_map. The only warning is to be careful when replacing any maps where we relied on the ordering.. prob not too common, but possible. We don't want to introduce any non-deterministic behavior

Reply:
And since we adopted c++11, we might as well take advantage of it.
Yes I think one of those changes caused all the residuals changes I'm seeing now, we were iterating over map items somewhere.
As for determinism I think unordered_map is with integer keys, as the hash function for those is the identity.